Feature/notification email

free2fqn opened the pull request:
Сделал нотификацию для емейла.

free2fqn committed:
add Spring Email, MailDTO, MailMapper, em@il credits on application.yml. started work on TextResendController

free2fqn committed:
add MailService, rename TextResendController to EmailResendControlle, working test-endpoint in EmailResendController, add config for yandex mail

free2fqn committed:
rename MailMapper->EmailMapper, MailService(Impl)->EmailService(Impl), MailResendController->EmailSendController

free2fqn committed:
commented field 'from' in MailDTO (don't know how to use it wisely). EmailMapper now takes mail field 'from' from environment. Deleted 'manual-send' end-point in EmailSendController, '/test' and '/mailDTO' endpoints work fine

nekromant322 commented on platform_dto/pom.xml:
```diff
@@ -19,5 +19,17 @@
             <groupId>org.projectlombok</groupId>
             <artifactId>lombok</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.jetbrains</groupId>
+            <artifactId>annotations</artifactId>
+            <version>RELEASE</version>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.jetbrains</groupId>
+            <artifactId>annotations</artifactId>
+            <version>RELEASE</version>
+            <scope>compile</scope>
+        </dependency>
```
зачем это?

free2fqn replied:
fixed

nekromant322 commented on platform_dto/src/main/java/dtos/MailDTO.java:
```diff
@@ -0,0 +1,35 @@
+//    @Nullable
+//    private String from;
+
+    @Nullable
+    private String replyTo;
+
+    @Nullable
+    private List<String> to;
+
+    @Nullable
+    private List<String> cc;
+
+    @Nullable
+    private List<String> bcc;
+
+    @Nullable
+    private Date sentDate;
+
+    @Nullable
+    private String subject;
+
+    @Nullable
+    private String text;
+}
```
@Nullable чего ради тут?  
а sentDate? у тебя реализована отложенная отправка? или спринг мэйл сам умеет отложенно отправлять?

free2fqn replied:
Действительно, ни @nullable, ни sentDate оказались не нужны

nekromant322 commented on platform_notification/src/main/java/com/override/controllers/EmailSendController.java:
```diff
@@ -0,0 +1,25 @@
+        emailService.sendSimpleMail(mailDTO);
+    }
+
+    @PostMapping("/test")
+    public void test() {
+        emailService.sendSimpleMail("free2fqn@gmail.com", "Test", "Test message");
+    }
```
зачем это мержить?

free2fqn replied:
fixed

nekromant322 commented on platform_notification/src/main/java/com/override/service/EmailService.java:
```diff
@@ -0,0 +1,16 @@
+
+    public void sendSimpleMail(MailDTO mailDTO);
+
+//    public void sendEmailWithAttachment(String toAddress, String subject, String message, String attachment) throws MessagingException, FileNotFoundException;
```
убрать

free2fqn replied:
fixed

nekromant322 commented on platform_notification/src/main/java/com/override/util/EmailMapper.java:
```diff
@@ -0,0 +1,28 @@
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.stereotype.Component;
+
+//TODO сейчас "From" достается из переменной-окружения. Но зачем-то же её можно регулировать, как настроить это?
```
избавиться от System.getenv() 
если что-то хочется конфигурить подобным образом, можно в ямле подтягивать переменные окружения и внедрять через @Value

free2fqn replied:
А @Value должна брать значение из yml или из переменной окружения? Или +- всё равно?
 Upd теперь берется из yml

nekromant322 commented on platform_notification/src/main/resources/application.yml:
```diff
@@ -1,10 +1,19 @@
+#TODO прочекать заходят ли переменные среды
+#TODO Надо зайти в гугл-аккаунт почты и включить настройку "less Secure Apps"
```
кому надо? почему в пул реквесте? создай задачку, если это нужно сделать позже не забыть  
если это должен сделать ты - делай, а тудушку убери)

free2fqn replied:
fixed

free2fqn committed:
clean up

free2fqn committed:
removed System.getEnv() from EmailMapper, now using @Value

free2fqn committed:
removed from MailDTO cc, bcc and sentDate.
Reworked EmailServiceImpl.sendSimpleMail(): now works with list of SimpleMailMessage, instead of one MailDTO; removed from EmailService overload .sendSimpleMail() for manual use.
EmailMapper.dtoToSimpleMail() replaced with dtoToListOfSimpleMails().
EmailMapper now doesn't write sentDate for email, not necessary.
